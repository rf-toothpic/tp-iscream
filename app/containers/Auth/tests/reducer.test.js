import authReducer, { initialState } from '../reducer'
import {
  updateField,
  assessorLoginRequest,
  assessorLoginRequestSuccess,
  assessorLoginRequestFailure,
  clearSession,
  defaultAction
} from '../actions'

describe('authReducer', () => {
  it('responds to default action', () => {
    expect(authReducer(undefined, defaultAction())).toMatchSnapshot()
  })
  it('returns the initial state', () => {
    expect(authReducer(undefined, {})).toMatchSnapshot()
  })

  it('handles the update field action', () => {
    expect(authReducer(undefined, updateField('email', 'test@test.com'))).toMatchSnapshot()
  })

  const initialStateLogin = initialState
  const afterLogin = { ...initialState, loading: true, error: null, emailError: null }
  it('handles login request', () => {
    expect(authReducer(initialStateLogin, assessorLoginRequest())).toEqual(afterLogin)
  })

  const data = { auth_token: '123123213' }
  it('handles login request success', () => {
    expect(
      authReducer(afterLogin, assessorLoginRequestSuccess({ authResponse: data }))
    ).toMatchSnapshot()
  })

  it('handles login request failure', () => {
    expect(
      authReducer(afterLogin, assessorLoginRequestFailure(new Error('badtime')))
    ).toMatchSnapshot()
  })

  it('clears data from the reducer on reload ', () => {
    expect(authReducer(afterLogin, clearSession())).toMatchSnapshot()
  })
})
