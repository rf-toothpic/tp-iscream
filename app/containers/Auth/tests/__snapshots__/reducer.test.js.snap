// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authReducer clears data from the reducer on reload  1`] = `
Object {
  "email": "",
  "emailError": null,
  "error": null,
  "loading": false,
  "password": "",
  "resetError": null,
  "resetPasswordError": null,
  "resetReqResponse": null,
  "resetResponse": null,
  "resetSuccess": false,
  "sessionData": Object {},
  "showInviteSuccess": false,
}
`;

exports[`authReducer handles login request failure 1`] = `
Object {
  "email": "",
  "emailError": null,
  "error": null,
  "loading": true,
  "password": "",
  "resetPasswordError": null,
  "resetReqResponse": null,
  "resetResponse": null,
  "resetSuccess": false,
  "sessionData": Object {},
  "showInviteSuccess": false,
}
`;

exports[`authReducer handles login request success 1`] = `
Object {
  "email": "",
  "emailError": null,
  "error": null,
  "loading": true,
  "password": "",
  "resetPasswordError": null,
  "resetReqResponse": null,
  "resetResponse": null,
  "resetSuccess": false,
  "sessionData": Object {
    "authResponse": Object {
      "auth_token": "123123213",
    },
  },
  "showInviteSuccess": false,
}
`;

exports[`authReducer handles the update field action 1`] = `
Object {
  "email": "test@test.com",
  "emailError": null,
  "error": null,
  "loading": false,
  "password": "",
  "resetPasswordError": null,
  "resetReqResponse": null,
  "resetResponse": null,
  "resetSuccess": false,
  "sessionData": Object {},
  "showInviteSuccess": false,
}
`;

exports[`authReducer responds to default action 1`] = `
Object {
  "email": "",
  "error": null,
  "loading": false,
  "password": "",
  "resetPasswordError": null,
  "resetReqResponse": null,
  "resetResponse": null,
  "resetSuccess": false,
  "sessionData": Object {},
  "showInviteSuccess": false,
}
`;

exports[`authReducer returns the initial state 1`] = `
Object {
  "email": "",
  "error": null,
  "loading": false,
  "password": "",
  "resetPasswordError": null,
  "resetReqResponse": null,
  "resetResponse": null,
  "resetSuccess": false,
  "sessionData": Object {},
  "showInviteSuccess": false,
}
`;
